### ISSUE TITLE
Define grammar for DSL (whole team)

### ISSUE THREAD

**Issue by r-e-stern** [uchicago-cs/chiventure#937]
We'll meet as a team to finalize ideas from our prototype from last week into a well-defined grammar for the new DSL that can be implemented by the group writing the grammar file.

**Comment by r-e-stern** [comment]
We met and decided that the grammar we implement for the first version of DSL should be very simple, easy to write, and lack most of DSL's core features, for the sake of producing a fully-fledged parser/lexer by the end of the sprint. This means the grammar will adhere to most of the 'simple' rules we defined in our [user stories](https://github.com/uchicago-cs/chiventure/wiki/DSL:-User-Stories) last week (#861).

**We should include the following 'advanced' features:**
- Whitespace-agnosticism

**We should wait to implement the following 'advanced' features**
- Actions
- Synonyms
- Newline-agnosticism
- Declaring an item without specifying its room
- Variables
- Default parameters (or, at least, implement them in the lexer rather than the parser)

That means the basic grammar should be as follows:

***

### Declaring a Thing (Game, Room, or Item)

**For a `GAME` (one per file), where `\n` represents a newline character:**
```
GAME START [room id 0] END [room id 1] \n
```
For example:
```
GAME FROM reg TO mansueto
```

**For a `ROOM`:**
```
ROOM [room id] \n
```
For example:
```
ROOM reg
```

**For an `ITEM` in a `ROOM`:**
```
ITEM [item id] IN [room id] \n
```
For example:
```
ITEM doing honest work in college IN reg
```

### Giving a Thing Properties

**All things should have properties, which can be declared thusly:**
```
[property]: "[value]"\n
```
For example:
```
ITEM doing honest work in college IN reg
             short: "A book."
long: "This is Doing Honest Work in College, baybee! It's the best book — nay, the only book — ever written!"
```
The parser should support both whitespace treatments, and any statement of `[property]: "[value]"` on one line.

### Ending a Thing

Things should be terminated... somehow. We'd prefer that be done automatically by the parser, but if that's not working out we could throw a little `;` or something (dealer's choice) after a thing's last property/value pairing. Please no HTML-style `/ITEM` tags; people will inevitably forget to use them,

***
...and that's it! We just want to get the parser working for now, we don't want to try to build a full-featured language in 10 days — [that's how you wind up with JavaScript](https://thenewstack.io/brendan-eich-on-creating-javascript-in-10-days-and-what-hed-do-differently-today/).

(JavaScript, you know I love you, please forgive me.)

**Comment by r-e-stern** [comment]
I have duplicated this documentation into the brand-new [DSL Grammar wiki page](https://github.com/uchicago-cs/chiventure/wiki/DSL:-DSL-Grammar), which we should keep continually updated as we add new features to the DSL.

Shower-thought-but-not-in-the-shower-because-I-keep-my-computer-away-from-water-at-all-costs: now that it's got a grammar, we should consider giving the DSL a proper name.

Closing the issue.
